lib/doc2pdf/document.rb

```diff
@@ -1,14 +1,17 @@
 # frozen_string_literal: true
 
 require "docx"
+require "uri"
 
 module Doc2pdf
-  # Wraps a Docx::Document.
+  # Wraps a ::Docx::Document.
   class Document
-    attr_reader :docx
+    attr_accessor :docx
 
     def initialize(file:)
-      self.docx = Docx::Document.open(file)
+      file = load_from_uri(file)
```

Comment:
As per my previous comment, I'd advise against this.
If you want to put this responsibility onto this gem, then we could extract the loading behaviour to a separate object but I don't think it's worth it.